### PR TITLE
Error handling and init no-copy in JNI

### PR DESCRIPTION
**Pull request by thomas-quadratic**
This PR has two purposes:

1. Error handling: as much as possible, pub extern fn should not panic. We catch errors and throw exceptions in the JVM.
2. No-copy: as much as possible, we should not have to copy data before passing to the FFI.

This is still a draft PR for the moment, as I am making sure tests are in sync.

### CODE REVIEW

File: ipa-multipoint/ipa_multipoint_jni/Cargo.toml
Comment (matkt) [inline ipa-multipoint/ipa_multipoint_jni/Cargo.toml:25]: I prefer to have a specific commit in order to build the same thing everytime

**Review by matkt — COMMENTED** [review]
LGTM, if we can just add a specific commit for rust verkle and fix javadoc issues before merging

It will be nice to add some description for each method in order to simplify the readability

File: ipa-multipoint/ipa_multipoint_jni/Cargo.toml
Comment (thomas-quadratic) [reply ipa-multipoint/ipa_multipoint_jni/Cargo.toml]: No problem, done.

**Review by matkt — APPROVED** [review]

**Pull request merged**
